File: sdk/android/story-sdk/src/main/kotlin/com/openstory/sdk/OpenStoryCallbacks.kt
package com.openstory.sdk

import com.openstory.sdk.model.OpenStoryAnalyticsEvent
import com.openstory.sdk.model.OpenStoryCtaPayload

interface OpenStoryCallbacks {
    fun onStoryBarImpression(event: OpenStoryAnalyticsEvent) = Unit

    fun onStoryGroupTap(event: OpenStoryAnalyticsEvent) = Unit

    fun onStoryView(event: OpenStoryAnalyticsEvent) = Unit

    fun onStoryComplete(event: OpenStoryAnalyticsEvent) = Unit

    fun onStoryCtaTap(payload: OpenStoryCtaPayload) = Unit

    fun onViewerClose(event: OpenStoryAnalyticsEvent) = Unit

    fun onGroupComplete(event: OpenStoryAnalyticsEvent) = Unit

    fun onError(placementKey: String, throwable: Throwable) = Unit
}

File: sdk/flutter/ios/OpenStorySDK/OpenStoryCallbacks.swift
import Foundation

@MainActor
public protocol OpenStoryCallbacks: AnyObject {
    func onStoryBarVisibilityChanged(placementKey: String, isVisible: Bool)
    func onStoryBarImpression(event: OpenStoryAnalyticsEvent)
    func onStoryGroupTap(event: OpenStoryAnalyticsEvent)
    func onStoryView(event: OpenStoryAnalyticsEvent)
    func onStoryComplete(event: OpenStoryAnalyticsEvent)
    func onStoryCtaTap(payload: OpenStoryCtaPayload)
    func onViewerClose(event: OpenStoryAnalyticsEvent)
    func onGroupComplete(event: OpenStoryAnalyticsEvent)
    func onError(placementKey: String, error: Error)
}

@MainActor
public extension OpenStoryCallbacks {
    func onStoryBarVisibilityChanged(placementKey: String, isVisible: Bool) {}
    func onStoryBarImpression(event: OpenStoryAnalyticsEvent) {}
    func onStoryGroupTap(event: OpenStoryAnalyticsEvent) {}
    func onStoryView(event: OpenStoryAnalyticsEvent) {}
    func onStoryComplete(event: OpenStoryAnalyticsEvent) {}
    func onStoryCtaTap(payload: OpenStoryCtaPayload) {}
    func onViewerClose(event: OpenStoryAnalyticsEvent) {}
    func onGroupComplete(event: OpenStoryAnalyticsEvent) {}
    func onError(placementKey: String, error: Error) {}
}
